src/js/client/editprefset-settingsTable-widget.js

```diff
@@ -0,0 +1,324 @@
+        },
+        enableProductListener: {
+            funcName: "gpii.devpmt.settingsTable.enableProductListener",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"] // event
+        },
+        updateTermUsage: {
+            funcName: "gpii.devpmt.settingsTable.updateTermUsage",
+            args: ["{that}"]
+        },
+        removeProduct: {
+            funcName: "gpii.devpmt.settingsTable.removeProduct",
+            args: ["{that}", "{gpii.devpmt.editPrefs}"]
+        }
+    },
+    markupEventBindings: {
+        enabledBooleanInputs: {
+            method: "click",
+            args: "{that}.enableProductListener"
+        },
+        addContextButton: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.events.openAddContextDialog.fire"
+        },
+        valueDisplayCell: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editValueEvent"
+        },
+        removeProductButton: {
+            method: "click",
+            args: "{that}.removeProduct"
+        },
+        pspShowCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspShow"
+        },
+        pspMemoryCheckboxes: {
+            method: "click",
+            args: "{gpii.devpmt.editPrefs}.editPspMemory"
+        }
+    },
+    listeners: {
+        "onCreate": [
+            {
+                "func": "{that}.updateLunrIndex"
+            },
+            {
+                "func": "{that}.filterInit"
+            },
+            {
+                "func": "{that}.updateTermUsage"
+            }
+        ],
+        "onMarkupRendered": [
+            {
+                "func": "{that}.filterSettings"
+            }
+        ]
+    },
+    lunrIndex: null
+});
+
+/**
+ * gpii.devpmt.settingsTable.filterInit - The users current filter options
+ * will be stored on the parent component and updated via model relay, such
+ * that they are saved if this component is destroyed and recreated.
+ * The first time it is created though, we want to set the default values,
+ * and trigger them via change applier so they are set on the parent.
+ *
+ * @param (Object) that
+ */
+gpii.devpmt.settingsTable.filterInit = function (that) {
+    if (that.model.settingsFilter === null) {
+        that.applier.change("settingsFilter", "");
+    }
+    if (that.model.allSettingsEnabled === null) {
+        that.applier.change("allSettingsEnabled", "mysettings");
+    }
+};
+
+gpii.devpmt.settingsTable.enableProductListener = function (that, devpmt, event) {
+    var checked = event.currentTarget.checked;
+    var context = event.currentTarget.dataset.context;
+    var product = event.currentTarget.dataset.product;
+
+    if (gpii.devpmt.prefsetsForApplication(devpmt.model.flatPrefs, product).length === 1 &&
+        checked === false) {
+        var appId = product.slice(38); // TODO Ontology lookup
+        devpmt.applier.change("activeModalDialog", {
```

Comment:
It will be good to share "removeProduct()" function for line 200-201.

Reply:
Thanks, this was a very use observation. I've changed the arguments for that method a bit so it can be used for both operations.